### PR TITLE
kms: add direct types for KMSKeyRing

### PR DESCRIPTION
**Pull request by justinsb**

### CODE REVIEW

File: apis/kms/v1beta1/kmskeyring.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	// +optional
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	//   // Output only. The resource name for the
+	// //  [KeyRing][google.cloud.kms.v1.KeyRing] in the format
+	// //  `projects/*/locations/*/keyRings/*`.
+	// Name *string `json:"name,omitempty"`
+
+	// The self link of the created KeyRing in the format projects/{project}/locations/{location}/keyRings/{name}.
+	SelfLink *string `json:"selfLink,omitempty"`
```
Comment (yuwenma) [inline apis/kms/v1beta1/kmskeyring.go:62]: or `externalRef`?

**Review by yuwenma — COMMENTED** [review]
/approve
/lgtm

/hold  One naming comment. feel free to unhold

File: apis/kms/v1beta1/kmskeyring.go (hunk 1 above)
Comment (justinsb) [reply apis/kms/v1beta1/kmskeyring.go:62]: So this PR is just creating go types for the existing structure, I don't think we want to change the schema.  Renaming selfLink -> externalRef might be considered a breaking change, but regardless it likely shouldn't happen in this PR (IMO)

**Comment by justinsb** [comment]
I think I've addressed the hold question, cancelling hold

/hold cancel

**Pull request merged**